src/main/scala/com/microsoft/azure/iot/iothubreact/scaladsl/IoTHubPartition.scala

```diff
@@ -33,7 +33,7 @@ object IoTHubPartition extends Logger {
   * @param partition IoT hub partition number (0-based). The number of
   *                  partitions is set during the deployment.
   */
-private[iothubreact] case class IoTHubPartition(val partition: Int) extends Logger {
+private[iothubreact] case class IoTHubPartition(val partition: Int)(implicit cfg: CPConfiguration = new CPConfiguration()) extends Logger {
```

Comment:
thanks for this, I really like DI via implicits 👍 
Following my other comment about `ICPConfiguration`, here and in other classes where DI is needed, please use `implicit config: ICPConfiguration` (without a default, i.e. set the implicit val in the calling classes)